generic3g/specs/FieldSpec.F90

```diff
@@ -181,7 +181,7 @@ function new_FieldSpec_geom(unusable, geom, vertical_grid, vertical_dim_spec, ty
 
       ! regrid_param
       field_spec%regrid_param = EsmfRegridderParam() ! use default regrid method
-      regrid_method = get_regrid_method_(field_spec%standard_name)
+      regrid_method = get_regrid_method_from_field_dict(field_spec%standard_name)
```

Comment:
```suggestion
```
1. Method can return an error so does not belong in a constructor.
2. The value should be passed explicitly in the interface (optional arg)

Reply:
So now it is
```Fortran
! regrid_param
field_spec%regrid_param = EsmfRegridderParam() ! use default regrid method
if (present(regrid_param)) field_spec%regrid_param = regrid_param
```
If not specified, it defaults to the `regrid_param` based on the default regridding method.